analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ParameterLoaderTest.cs

```diff
@@ -213,5 +221,24 @@ private static SonarCompilationReportingContext CreateCompilationWithOption(stri
             var compilationContext = new CompilationAnalysisContext(compilation, options, _ => { }, _ => true, default);
             return new(AnalysisScaffolding.CreateSonarAnalysisContext(), compilationContext);
         }
+        private string GenerateSonarLintWithParametrizedRule(string ruleId, string key, string value)
+        {
+            var ruleParameters = new List<SonarLintXmlRule>()
+            {
+                new SonarLintXmlRule()
+                {
+                    Key = ruleId,
```

Comment:
```suggestion

```

Reply:
Can you explain this one better? The Key on the SonarLintXmlRule refers to the rule for which you'll pass all the parameters as key/value pairs (es. "S1234").

Reply:
Ok I re-saw this and it looks confusing at first glance but now I understand what happens.  🤦 
You are setting two `Keys` not one. 

Ignore this comments please.